src/factories/calorimetry/CalorimeterClusterRecoCoG_factory.h

```diff
@@ -0,0 +1,62 @@
+// Copyright 2023, Wouter Deconinck
+// Subject to the terms in the LICENSE file found in the top-level directory.
+//
+
+#pragma once
+
+#include <algorithms/calorimetry/CalorimeterClusterRecoCoG.h>
+#include <services/geometry/dd4hep/JDD4hep_service.h>
+#include <extensions/jana/JOmniFactory.h>
+
+
+namespace eicrecon {
+
+class CalorimeterClusterRecoCoG_factory : public JOmniFactory<CalorimeterClusterRecoCoGConfig> {
+
+    const dd4hep::Detector* m_detector;
+    eicrecon::CalorimeterClusterRecoCoG m_algo;
+
+public:
+    void Init() override {
+
+        DeclarePodioOutput<edm4eic::Cluster>(GetOutputTags()[0]);
+        DeclarePodioOutput<edm4eic::MCRecoClusterParticleAssociation>(GetOutputTags()[1]);
+
+        DeclareParameter("samplingFraction", GetConfig().sampFrac);
```

Comment:
One thing I haven't figured out yet is how we aim to allow algorithms to define their config options so we don't have to duplicate them in the factory. How does this approach get us closer to that? A generic factory will likely have to take the algorithm class as a template argument, and we'll have to use that algorithm's information to build a list of `DeclareParameters` with another fold expression? It won't work with the current struct approach in the algorithms, that's for sure (it doesn't have a way to get 
names, for starters). Maybe we should just propagate the algorithms property interface already. Those are mature enough.

Reply:
Just brainstorming here: Probably by using a templated visitor pattern. Consider: 
```c++

class ConfigT {
    int bucket_count = 2;
    float threshold = 6.7;
}

template <>
void Configure<ConfigT>(ParameterRegistrar& visitor, ConfigT& config) {
    visitor.RegisterParameter<int>("buckets", &config.bucket_count);
    visitor.RegisterParameter<float>("threshold", &config.threshold);
}
```

We then have separate subclasses of `ParameterRegistrar` for JANA, Gaudi, Art, etc.

Reply:
The other approach, in algorithms ([`rroperty.h`](https://github.com/eic/algorithms/blob/main/core/include/algorithms/property.h#L73), [`ClusterRecoCoG`](https://github.com/eic/algorithms/blob/main/calorimetry/include/algorithms/calorimetry/ClusterRecoCoG.h#L54)), is to have algorithms define properties which get registered internally. This can then be queried by the framework.

Reply:
Yeah, we are also using that pattern in JANA in `JEventProcessorSequential.h`. I have mixed feelings about it. The major pluses I see (compared to the visitor approach) are:
1. We can change the implementation at a very deep level without interfering with user's code
2. There is very little surface area for people to get confused
 
The major minuses I see are:
1. We no longer have a POD
2. If something goes wrong we are going to have to reason very carefully about C++ member initialization order

I would usually reach for a visitor pattern first just based off personal preference, but the more I think this one through the better it looks. I can imagine a JOmniFactory that represents collection types and Services the same way, and this could be a good first step to making JOmniFactory fully generic. I'll give it a whirl.